scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,335 @@
+          || isStringBlankOrNull(resultFolder)
+          || isStringBlankOrNull(containerSerialNumber)
+          || isStringBlankOrNull(poolName)) {
+        return;
+      }
+      // Container represents one SMRT Cell
+      Map<String, String> containerInfo = new HashMap<>();
+      containerInfo.put("ResultFolder", resultFolder);
+      containerInfo.put("Position", position);
+      containerInfo.put("ContainerSerialNumber", containerSerialNumber);
+      containerInfo.put("PoolName", poolName);
+
+      // Add container to SMRT cell positionList
+      List<Map<String, String>> tempPositionList = dto.getPositionList();
+      if (dto.getPositionList() == null) {
+        tempPositionList = new ArrayList<>();
+      }
+      tempPositionList.add(containerInfo);
+      dto.setPositionList(tempPositionList);
+    };
+  }
+
+  /**
+   * Extract a string expression from the metadata file and write it into the DTO.
+   *
+   * @param expression the XPath expression yielding the string
+   * @param setter writer for the string
+   * @return
+   */
+  private static RevioPacBioProcessor.ProcessMetadata processString(
+      String expression, BiConsumer<PacBioNotificationDto, String> setter) {
+    XPathExpression expr = RunProcessor.compileXPath(expression)[0];
+    return (document, dto, timeZone) -> {
+      String result = (String) expr.evaluate(document, XPathConstants.STRING);
+      if (result != null) {
+        setter.accept(dto, result);
+      }
+    };
+  }
+
+  public RevioPacBioProcessor(Builder builder, String s) {
+    super(builder);
+  }
+
+  public static RunProcessor create(Builder builder, ObjectNode parameters) {
+    JsonNode address = parameters.get("address");
+    return address.isTextual()
+        ? new RevioPacBioProcessor(builder, address.textValue().replaceAll("/+$", ""))
+        : null;
+  }
+
+  @Override
+  public Stream<File> getRunsFromRoot(File root) {
+    return Arrays.stream(
+        root.listFiles(f -> f.isDirectory() && RUN_DIRECTORY.matcher(f.getName()).matches()));
+  }
+
+  @Override
+  public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
+    // We create one DTO for a run, but there are going to be many wells with
+    // independent and
+    // duplicate metadata that will simply
+    // overwrite in the shared DTO. If the data differs, the last well wins.
+    PacBioNotificationDto dto = new PacBioNotificationDto();
+    dto.setPairedEndRun(false);
+    dto.setSequencerFolderPath(runDirectory.getAbsolutePath());
+
+    // Get the number of SMRT Cells in the run directory
+    dto.setLaneCount((int) getCellDirectory(runDirectory).count());
+
+    // Presence of Transfer_Test_*.txt indicates the run has started, get the creation time
+    getMetadataDirectory(runDirectory)
+        .flatMap(
+            metadataDirectory ->
+                Arrays.stream(
+                    metadataDirectory.listFiles(file -> TRANSFER_TEST.test(file.getName()))))
```

Comment:
I think I'm missing something here, but are you trying to get a list of files and then filter them based on whether they match the TRANSFER_TEST pattern? And if so, would you consider doing it as a separate `flatMap(get list of files).filter(TRANSFER_TEST.test(...))` for clarity?